apps/api/src/domain/services/query-services/base-query.service.ts

```diff
@@ -110,7 +101,7 @@ export abstract class BaseQueryService<
     public async fetchById(
         id: unknown,
         userWithGroups?: CoscradUserWithGroups
-    ): Promise<ResultOrError<Maybe<AggregateByIdQueryResult<ViewModelWithTags<UViewModel>>>>> {
+    ): Promise<ResultOrError<Maybe<IDetailQueryResult<ViewModelWithTags<UViewModel>>>>> {
```

Comment:
Yea. I suppose that's because this is at the boundary of the system. Let me break it down.
- Promise: the result is async
- ResultOrError: the result could be an `InternalError` instance instead
- Maybe: the result could be our internal `NotFound` symbol
- AggregateByIdQueryResult: The view model is combined with available actions in our standard format (if found and no error)
- UViewModel: generic to the type of the specific aggregate view model we are returning

Reply:
Great, thanks for this.  I figured it went something like that.